### DISCUSSION TITLE
Why doesn't Quarkus resolve a static resource when I replace path in a preMatching filter?

### DISCUSSION THREAD

**Discussion by okarmazin** [quarkusio/quarkus#43027] [Q&A]
I know this is likely intended behavior, hence not opening a ticket but a Q/A.

When a request comes in, I need to try and serve a static HTML file of the same name. My first try is to just add `.html` at the and of the path in a prematching filter.

However, Quarkus still resolves the old path without the `.html` suffix.

Here's the filter (simplified):
```
@ServerRequestFilter(preMatching = true)
fun htmlRerouter(request: ContainerRequestContext) {
    when {
        ....
        else -> {
            val oldUri = request.uriInfo.requestUri
            val newUri = request.uriInfo.requestUriBuilder.replacePath(oldUri.path + ".html").build()
            request.setRequestUri(newUri)
            Log.info("changing URI:")
            Log.info("FROM: $oldUri")
            Log.info("TO: ${request.uriInfo.requestUri}")
        }
    }
}
```

Log Output:

```
changing URI:
FROM: http://localhost:8080/admin/dashboard
TO: http://localhost:8080/admin/dashboard.html
"GET /admin/dashboard HTTP/1.1" 404 243214
```

Why is Quarkus telling me the path is still `/dashboard` when I replaced it with `/dashboard.html`?

**Comment by phillip-kruger** [comment]
Does that static file exist ? Also I think you will need to do a forward or redirect.

**Reply by okarmazin** [reply]
Yes, it exists. Navigating directly to the url `/admin/dashboard.html` serves the file.

Note I already solved my problem by implementing that route and loading the file from resources.

**Reply by okarmazin** [reply]
I am really just curious as to why changing the request path before matching doesn't work. It may be working as intended, but I don't know why.
Are static resources resolved even before the filter, and after the filter only application routes are matched? etc.

**Reply by phillip-kruger** [reply]
I think by the time the filter run, the matching of paths already happened and the correct resolver has been loaded. That is why I say you need to do a redirect or forward, so that matching happens again.
